assets/js/directives/on-click-copy.directive.js

```diff
@@ -0,0 +1,43 @@
+
+angular
+  .module('walletApp')
+  .directive('onClickCopy', onClickCopy);
+
+function onClickCopy ($window, $document) {
+  const directive = {
+    restrict: 'A',
+    scope: false,
+    link: link
+  };
+  return directive;
+
+  function link (scope, elem, attrs) {
+    scope.copy = () => {
+      let text = scope.$eval(attrs.onClickCopy);
+
+      if (!angular.isString(text) || text.length === 0) {
+        return;
+      }
+
+      let textElem = angular.element(`<span>${text}</span>`);
+      textElem.css({ position: 'absolute', left: '-1000px' });
+      angular.element($document[0].body).append(textElem);
```

Comment:
What do you mean by normal template? We need to create a dom node that's out of view, so that we can tell the browser to copy its text. Not sure if we can do that with a conventional directive.